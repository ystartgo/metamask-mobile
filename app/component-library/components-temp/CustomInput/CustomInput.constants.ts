export const TICKER = 'DAI';
export const MAX_VALUE = '200.12';
export const DEFAULT_VALUE =
  '115792089237316195423570985008687907853269984665640564039457.584007913129639936';
