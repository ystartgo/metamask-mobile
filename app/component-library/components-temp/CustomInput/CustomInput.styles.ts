// Third party dependencies.
import { StyleSheet } from 'react-native';

import { Theme } from '../../../util/theme/models';
/**
 * Style sheet for Custom Input component.
 *
 * @returns StyleSheet object.
 */

const styleSheet = (params: { theme: Theme }) => {
  const { theme } = params;
  return StyleSheet.create({
    container: {
      backgroundColor: theme.colors.background.default,
      margin: 10,
      borderRadius: 5,
      padding: 10,
      flexDirection: 'row',
      justifyContent: 'space-between',
    },
    body: {
      flexDirection: 'row',
      marginHorizontal: 5,
      width: '90%',
      alignItems: 'center',
    },
    input: {
      marginHorizontal: 3,
      marginVertical: 5,
      paddingTop: 0,
      paddingBottom: 0,
      flexDirection: 'row',
    },
    maxValueText: {
      fontSize: 14,
      color: theme.colors.text.alternative,
    },
    ticker: {
      paddingLeft: 5,
    },
    warningValue: {
      color: theme.colors.error.default,
    },
  });
};

export default styleSheet;
