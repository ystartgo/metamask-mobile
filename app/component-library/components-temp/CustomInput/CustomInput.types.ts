export interface CustomInputProps {
  /**
   * Token native symbol
   * @default 'ETH'
   */
  ticker: string;
  /**
   * Function to be called on input change
   */
  getUpdatedValue: (value: string) => void;
  /**
   * Maximum available value. Token balance.
   */
  maxAvailableValue: string;
  /**
   * Defined value of the input
   */
  defaultValue?: string;
}
