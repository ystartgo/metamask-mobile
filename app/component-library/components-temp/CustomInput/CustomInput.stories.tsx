// Third party dependencies.
import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Internal dependencies.
import CustomInput from './CustomInput';
import { TICKER, MAX_VALUE, DEFAULT_VALUE } from './CustomInput.constants';

const getUpdatedValue = (value: string) => {
  // do something with the value
  /* eslint-disable no-console */
  console.log('value', value);
};

storiesOf('Component Library / CustomInput', module)
  .add('Default', () => (
    <CustomInput
      ticker={TICKER}
      getUpdatedValue={getUpdatedValue}
      maxAvailableValue={MAX_VALUE}
    />
  ))
  .add('With default value', () => (
    <CustomInput
      ticker={TICKER}
      getUpdatedValue={getUpdatedValue}
      maxAvailableValue={MAX_VALUE}
      defaultValue={DEFAULT_VALUE}
    />
  ));
