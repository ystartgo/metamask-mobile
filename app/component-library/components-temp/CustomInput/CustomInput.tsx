// Third party dependencies.
import React, { useEffect, useState } from 'react';
import { TextInput, Text, View } from 'react-native';

// External dependencies.
import { useStyles } from '../../hooks';
import formatNumber from '../../../util/formatNumber';
import { strings } from '../../../../locales/i18n';

// Internal dependencies.
import { CustomInputProps } from './CustomInput.types';
import stylesheet from './CustomInput.styles';

const CustomInput = ({
  ticker,
  maxAvailableValue,
  defaultValue,
  getUpdatedValue,
}: CustomInputProps) => {
  const [value, onChangeValue] = useState(defaultValue || '');
  const [isDefaultValuePassed, setIsDefaultValuePassed] = useState(false);

  useEffect(() => {
    getUpdatedValue(value);
    setIsDefaultValuePassed(false);
  }, [value, getUpdatedValue]);

  useEffect(() => {
    if (defaultValue) {
      setIsDefaultValuePassed(true);
    }
  }, [defaultValue]);

  const showMaxValue = () => {
    onChangeValue(maxAvailableValue);
  };

  const { styles } = useStyles(stylesheet, {});

  return (
    <View style={styles.container}>
      <View style={styles.body}>
        {isDefaultValuePassed ? (
          <Text style={styles.warningValue}>
            <Text>{`${formatNumber(value)} ${ticker}`}</Text>
          </Text>
        ) : (
          <TextInput
            multiline
            onChangeText={onChangeValue}
            value={formatNumber(value)}
            placeholder={`Enter a number here (${ticker})`}
            keyboardType="numeric"
            style={[styles.input]}
          />
        )}
      </View>
      {!isDefaultValuePassed && (
        <Text style={styles.maxValueText} onPress={showMaxValue}>
          {strings('input.max_value')}
        </Text>
      )}
    </View>
  );
};

export default CustomInput;
