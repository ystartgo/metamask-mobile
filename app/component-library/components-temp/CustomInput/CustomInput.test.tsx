import React from 'react';
import { shallow } from 'enzyme';
import CustomInput from './CustomInput';
import { CustomInputProps } from './CustomInput.types';
import { TICKER, MAX_VALUE, DEFAULT_VALUE } from './CustomInput.constants';

describe('CustomInput', () => {
  let props: CustomInputProps;

  beforeEach(() => {
    props = {
      ticker: TICKER,
      getUpdatedValue: jest.fn(),
      maxAvailableValue: MAX_VALUE,
      defaultValue: DEFAULT_VALUE,
    };
  });

  const renderComponent = () => shallow(<CustomInput {...props} />);

  it('should render correctly', () => {
    const component = renderComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when defaultValue is not provided', () => {
    props.defaultValue = undefined;
    const component = renderComponent();
    expect(component).toMatchSnapshot();
  });
});
