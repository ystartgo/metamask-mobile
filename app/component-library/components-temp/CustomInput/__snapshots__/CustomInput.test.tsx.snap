// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 5,
        "width": "90%",
      }
    }
  >
    <TextInput
      keyboardType="numeric"
      multiline={true}
      onChangeText={[Function]}
      placeholder="Enter a number here (DAI)"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingBottom": 0,
            "paddingTop": 0,
          },
        ]
      }
      value="115,792,089,237,316,195,423,570,985,008,687,907,853,269,984,665,640,564,039,457.584,007,913,129,639,936"
    />
  </View>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#535A61",
        "fontSize": 14,
      }
    }
  >
    Max
  </Text>
</View>
`;

exports[`CustomInput should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 5,
        "width": "90%",
      }
    }
  >
    <TextInput
      keyboardType="numeric"
      multiline={true}
      onChangeText={[Function]}
      placeholder="Enter a number here (DAI)"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingBottom": 0,
            "paddingTop": 0,
          },
        ]
      }
      value="115,792,089,237,316,195,423,570,985,008,687,907,853,269,984,665,640,564,039,457.584,007,913,129,639,936"
    />
  </View>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#535A61",
        "fontSize": 14,
      }
    }
  >
    Max
  </Text>
</View>
`;

exports[`CustomInput should render correctly when defaultValue is not provided 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 5,
        "width": "90%",
      }
    }
  >
    <TextInput
      keyboardType="numeric"
      multiline={true}
      onChangeText={[Function]}
      placeholder="Enter a number here (DAI)"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingBottom": 0,
            "paddingTop": 0,
          },
        ]
      }
      value=""
    />
  </View>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#535A61",
        "fontSize": 14,
      }
    }
  >
    Max
  </Text>
</View>
`;
